src/utils/colors-map.spec.ts

```diff
@@ -0,0 +1,188 @@
+import ColorMap, { getRGBValueFromColorObject, makeRGBObjectFromColor } from './colors-map';
+
+const DEFAULT_COLOR = [255, 255, 255];
+const DEFAULT_HIGLIGHT_COLOR = [0, 100, 255];
+
+describe("Color Map", () => {
+  test("Should create color Map", () => {
+    const colorMap = new ColorMap();
+
+    expect(colorMap).toBeDefined();
+    expect(colorMap.randomColorMap).toBeDefined();
+    expect(colorMap.colorMap).toBeDefined();
+  });
+
+  test("Should return default color", () => {
+    const colorMap = new ColorMap();
+
+    const tile = {
+      id: 'test'
+    };
+
+    const randomColor = colorMap.getColor(tile, {});
+    expect(randomColor).toEqual(DEFAULT_COLOR);
+  });
+
+  test("Should return default color", () => {
+    const colorMap = new ColorMap();
+
+    const tile = {
+      id: 'test'
+    };
+
+    const randomColor = colorMap.getColor(tile, {});
+    expect(randomColor).toEqual(DEFAULT_COLOR);
+
+    const defaultColor = colorMap.getColor(tile, { coloredBy: 0 });
+    expect(defaultColor).toEqual(DEFAULT_COLOR);
+  });
+
+  test("Should return random color", () => {
+    const colorMap = new ColorMap();
+
+    const tile = {
+      id: 'test'
+    };
+
+    const randomColor = colorMap.getColor(tile, { coloredBy: 1 });
+    expect(randomColor).toBeDefined();
+    const colorForTheSameTile = colorMap.getColor(tile, { coloredBy: 1 });
+
+    expect(randomColor).toEqual(colorForTheSameTile);
+  });
+
+  test("Should return color by depth", () => {
+    const colorMap = new ColorMap();
+
+    const tile = {
+      id: 'test',
+      depth: 1
+    };
+
+    const colorByDepth = colorMap.getColor(tile, { coloredBy: 2 });
+    expect(colorByDepth).toBeDefined();
+    const colorForTheSameTile = colorMap.getColor(tile, { coloredBy: 2 });
+
+    expect(colorByDepth).toEqual(colorForTheSameTile);
+  });
+
+  test("Should return DEFAULT color in _getColorByTile method if colorMap isEmpty", () => {
+    const colorMap = new ColorMap();
+
+    const tile = {
+      id: 'test'
+    };
```

Comment:
You could initialize `colorMap` and `tile` in `beforeEach` hook

Reply:
Done